src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -308,29 +308,53 @@ async Task<T> IDurableOrchestrationContext.CreateTimer<T>(DateTime fireAt, T sta
         {
             this.ThrowIfInvalidAccess();
 
+            DateTime intervalFireAt = fireAt;
+            TimeSpan longRunningTimerIntervalLength = TimeSpan.FromDays(3);
+
             if (!this.durabilityProvider.ValidateDelayTime(fireAt.Subtract(this.InnerContext.CurrentUtcDateTime), out string errorMessage))
             {
-                throw new ArgumentException(errorMessage, nameof(fireAt));
+                intervalFireAt = this.InnerContext.CurrentUtcDateTime.AddDays(longRunningTimerIntervalLength.TotalDays);
             }
 
-            this.IncrementActionsOrThrowException();
-            Task<T> timerTask = this.InnerContext.CreateTimer(fireAt, state, cancelToken);
+            T result = default;
 
-            this.Config.TraceHelper.FunctionListening(
-                this.Config.Options.HubName,
-                this.OrchestrationName,
-                this.InstanceId,
-                reason: $"CreateTimer:{fireAt:o}",
-                isReplay: this.InnerContext.IsReplaying);
+            while (this.InnerContext.CurrentUtcDateTime < fireAt && !cancelToken.IsCancellationRequested)
+            {
+                this.IncrementActionsOrThrowException();
+                Task<T> timerTask = this.InnerContext.CreateTimer(intervalFireAt, state, cancelToken);
 
-            T result = await timerTask;
+                this.Config.TraceHelper.FunctionListening(
```

Comment:
I suggest we move this trace outside of the `while` loop so that we only trace once. The fact that we use multiple messages should probably not be exposed at this layer of logs.